validator/remote/src/main/java/tech/pegasys/teku/validator/remote/apiclient/OkHttpValidatorRestApiClient.java

```diff
@@ -0,0 +1,259 @@
+    post(SEND_SIGNED_AGGREGATE_AND_PROOF, signedAggregateAndProof, null);
+  }
+
+  @Override
+  public void subscribeToBeaconCommitteeForAggregation(
+      final int committeeIndex, final UInt64 aggregationSlot) {
+    final SubscribeToBeaconCommitteeRequest request =
+        new SubscribeToBeaconCommitteeRequest(committeeIndex, aggregationSlot);
+    post(SUBSCRIBE_TO_COMMITTEE_FOR_AGGREGATION, request, null);
+  }
+
+  @Override
+  public void subscribeToPersistentSubnets(final Set<SubnetSubscription> subnetSubscriptions) {
+    post(SUBSCRIBE_TO_PERSISTENT_SUBNETS, subnetSubscriptions, null);
+  }
+
+  public <T> Optional<T> get(
+      final ValidatorApiMethod apiMethod,
+      final Map<String, String> queryParams,
+      final Class<T> responseClass) {
+    final HttpUrl.Builder httpUrlBuilder = urlBuilder(apiMethod);
+    if (queryParams != null && !queryParams.isEmpty()) {
+      queryParams.forEach(httpUrlBuilder::addQueryParameter);
+    }
+
+    final Request request = new Request.Builder().url(httpUrlBuilder.build()).build();
+    return executeCall(request, responseClass);
+  }
+
+  private <T> Optional<T> post(
+      final ValidatorApiMethod apiMethod,
+      final Object requestBodyObj,
+      final Class<T> responseClass) {
+    final String requestBody;
+    try {
+      requestBody = jsonProvider.objectToJSON(requestBodyObj);
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException(e);
+    }
+
+    final Request request =
+        new Request.Builder()
+            .url(urlBuilder(apiMethod).build())
+            .post(RequestBody.create(requestBody, APPLICATION_JSON))
+            .build();
+
+    return executeCall(request, responseClass);
+  }
+
+  private HttpUrl.Builder urlBuilder(final ValidatorApiMethod apiMethod) {
+    return new HttpUrl.Builder()
+        .scheme(baseEndpoint.scheme())
+        .host(baseEndpoint.host())
+        .port(baseEndpoint.port())
+        .encodedPath(apiMethod.getPath());
+  }
+
+  private <T> Optional<T> executeCall(final Request request, final Class<T> responseClass) {
+    try (final Response response = httpClient.newCall(request).execute()) {
+      LOG.trace("{} {} {}", request.method(), request.url(), response.code());
+
+      switch (response.code()) {
+        case 200:
+          {
+            final String responseBody = response.body().string();
+            if (responseClass != null) {
+              final T responseObj = jsonProvider.jsonToObject(responseBody, responseClass);
+              return Optional.of(responseObj);
+            } else {
+              return Optional.empty();
+            }
+          }
+        case 202:
+        case 204:
+        case 404:
+        case 503:
+          {
+            return Optional.empty();
+          }
+        case 400:
+          {
+            LOG.warn("Invalid params response from Beacon Node API - {}", response.body().string());
```

Comment:
This feels like an error more than a warn - likely we've just missed performing some validator duty.